compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5719,11 +5719,18 @@ TR::Register* OMR::X86::TreeEvaluator::vectorBinaryArithmeticEvaluator(TR::Node*
 
 
    TR::InstOpCode nativeOpcode = getNativeSIMDOpcode(opcode, type, useRegMemForm);
+   OMR::X86::Encoding simdEncoding;
 
-   if (useRegMemForm && nativeOpcode.getMnemonic() == TR::InstOpCode::bad)
+   if (nativeOpcode.getMnemonic() != TR::InstOpCode::bad)
+      {
+      simdEncoding = nativeOpcode.getSIMDEncoding(&cg->comp()->target().cpu, type.getVectorLength());
+      }
+
+   if (useRegMemForm && (simdEncoding == OMR::X86::Legacy || nativeOpcode.getMnemonic() == TR::InstOpCode::bad))
```

Comment:
I think it would be helpful to add a comment indicating why the register-memory form cannot be used if `simdEncoding == OMR::X86::Legacy`.